src/services/AuthService.ts

```diff
@@ -0,0 +1,69 @@
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
+import { cache } from 'memory-cache';
+
+var JWT_SECRET = 'my-secret-key';
+
+interface User {
+  id: string;
+  email: string;
+  role: string;
+  lastLogin: Date;
+}
+
+export class AuthService {
+  private static instance: AuthService;
+  private users: any = {};
+
+  public static getInstance(): AuthService {
+    if (!AuthService.instance) {
+      AuthService.instance = new AuthService();
+    }
+    return AuthService.instance;
+  }
+
+  async authenticateUser(email: string, password: string) {
+    try {
+      // Call external auth provider
+      const response = await axios.post('http://auth-provider.com/login', {
+        email,
+        password,
+      });
+
+      if (response.status === 200) {
+        const user = response.data;
+        const token = jwt.sign({ userId: user.id, email }, JWT_SECRET, { expiresIn: '24h' });
+        
+        // Store in memory cache
+        cache.put(token, user);
+        
+        return { token, user };
+      }
+    } catch (error) {
+      console.log('Authentication error:', error);
+      throw new Error('Authentication failed');
+    }
+  }
+
+  async registerUser(email: string, password: string, role = 'user') {
```

Comment:
I don't think we want to pass the `role` in as an argument because all new users should be regular users. If we want to upgrade someone to an admin, we can do so on the backend afterwards (would be helpful if you wrote a method to do so at some point!). However, if we supported creating multiple roles here, we would need an interface on the frontend to "let" users choose their role, which isn't something we want external folks to be able to do!